lib/src/migrators/module.dart

```diff
@@ -325,41 +309,75 @@ class _ModuleMigrationVisitor extends MigrationVisitor {
 
     // Create a `@forward` rule for each dependency that has members that should
     // be forwarded.
-    var forwards = <String>[];
+    var loadPathForwards = <String>[];
+    var relativeForwards = <String>[];
     for (var url in shown.keys) {
       var hiddenCount = hidden[url]?.length ?? 0;
-      var forward = '@forward "${_absoluteUrlToDependency(url)}"';
+      var tuple = _absoluteUrlToDependency(url);
+      var forward = '@forward "${tuple.item1}"';
+      var isRelative = tuple.item2;
 
       // When not all members from a dependency should be forwarded, use a
       // `hide` clause to hide the ones that shouldn't.
       if (hiddenCount > 0) {
         var hiddenMembers = hidden[url].toList()..sort();
         forward += ' hide ${hiddenMembers.join(", ")}';
       }
-      forward += '$_semicolonIfNotIndented\n';
-      forwards.add(forward);
+      if (_possibleForwardConversions.containsKey(url)) {
+        _possibleForwardConversions[url].replacement = forward;
+      } else {
+        forward += '$_semicolonIfNotIndented\n';
+        (isRelative ? relativeForwards : loadPathForwards).add(forward);
+      }
     }
-    forwards.sort();
-    return forwards.join('');
+    loadPathForwards.sort();
+    relativeForwards.sort();
+    var forwards = loadPathForwards.followedBy(relativeForwards);
```

Comment:
```suggestion
    var forwards = [...loadPathForwards, ...relativeForwards];
```

It's generally a good idea to create a `List` rather than an `Iterable` if you're going to access it multiple times, since otherwise you may do lazy operations twice.

Reply:
Done